pkg/validation/app.go

```diff
@@ -16,14 +16,15 @@ import (
 )
 
 const (
-	catalogNotFoundTemplate         = "catalog %#q not found"
-	nameTooLongTemplate             = "name %#q is %d chars and exceeds max length of %d chars"
-	nameNotFoundReasonTemplate      = "name is not specified for %s"
-	targetNamespaceNotAllowed       = "target namespace %s is not allowed for in-cluster apps"
-	namespaceNotFoundReasonTemplate = "namespace is not specified for %s %#q"
-	labelInvalidValueTemplate       = "label %#q has invalid value %#q"
-	labelNotFoundTemplate           = "label %#q not found"
-	resourceNotFoundTemplate        = "%s %#q in namespace %#q not found"
+	catalogNotFoundTemplate           = "catalog %#q not found"
+	nameTooLongTemplate               = "name %#q is %d chars and exceeds max length of %d chars"
+	nameNotFoundReasonTemplate        = "name is not specified for %s"
+	targetNamespaceNotAllowedTemplate = "target namespace %s is not allowed for in-cluster apps"
+	namespaceNotFoundReasonTemplate   = "namespace is not specified for %s %#q"
+	labelsConflictTemplate            = "label %#q is not allowed with the %#q label"
```

Comment:
So far this is used only when both, the `giantswarm.io/cluster` and the `app-operator.giantswarm.io/version`, labels are used. I made this name general for possible future use cases, but feel free to correct it to be more accurate.